src/test/rustdoc-ui/intra-doc/disambiguator-mismatch.stderr

```diff
@@ -2,94 +2,138 @@ error: incompatible link kind for `S`
   --> $DIR/disambiguator-mismatch.rs:14:14
    |
 LL | /// Link to [struct@S]
-   |              ^^^^^^^^ help: to link to the enum, prefix with `enum@`: `enum@S`
+   |              ^^^^^^^^
    |
 note: the lint level is defined here
   --> $DIR/disambiguator-mismatch.rs:1:9
    |
 LL | #![deny(rustdoc::broken_intra_doc_links)]
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = note: this link resolved to an enum, which is not a struct
+help: to link to the enum, prefix with `enum@`
+   |
+LL | /// Link to [enum@S]
+   |              ^^^^^
 
 error: incompatible link kind for `S`
   --> $DIR/disambiguator-mismatch.rs:19:14
    |
 LL | /// Link to [mod@S]
-   |              ^^^^^ help: to link to the enum, prefix with `enum@`: `enum@S`
+   |              ^^^^^
    |
    = note: this link resolved to an enum, which is not a module
+help: to link to the enum, prefix with `enum@`
+   |
+LL | /// Link to [enum@S]
+   |              ^^^^^
 
 error: incompatible link kind for `S`
   --> $DIR/disambiguator-mismatch.rs:24:14
    |
 LL | /// Link to [union@S]
-   |              ^^^^^^^ help: to link to the enum, prefix with `enum@`: `enum@S`
+   |              ^^^^^^^
    |
    = note: this link resolved to an enum, which is not a union
```

Comment:
Can we make this `note` the primary span?

Reply:
What do you mean?

Reply:
Produce the following output by changing `err.note(&msg)` with `err.span_label(span, &msg)`:

```
error: incompatible link kind for `S`
  --> $DIR/disambiguator-mismatch.rs:24:14
   |
LL | /// Link to [union@S]
   |              ^^^^^^^ this link resolved to an enum, which is not a union
```